apache2.yaml

```diff
@@ -129,6 +129,29 @@ subpackages:
           mv "${{targets.destdir}}"/usr/share/apache2/icons "${{targets.subpkgdir}}"/usr/share/apache2/
           mv "${{targets.destdir}}"/usr/share/apache2/error "${{targets.subpkgdir}}"/usr/share/apache2/
 
+  - name: ${{package.name}}-oci-entrypoint
+    description: "OCI entrypoint for apache2"
+    dependencies:
+      runtime:
+        - ${{package.name}}
+        - dash-binsh
+    pipeline:
+      - runs: |
+          set -x
+          # Add the entrypoint
+          mkdir -p ${{targets.subpkgdir}}/usr/bin
+          mv entrypoint.sh ${{targets.subpkgdir}}/usr/bin/entrypoint.sh
+          chmod 0755 ${{targets.subpkgdir}}/usr/bin/entrypoint.sh
+          # Modify the default config
+          sed -ri \
+            -e 's!^(\s*CustomLog)\s+\S+!\1 /proc/self/fd/1!g' \
+            -e 's!^(\s*ErrorLog)\s+\S+!\1 /proc/self/fd/2!g' \
+            -e 's!^(\s*TransferLog)\s+\S+!\1 /proc/self/fd/1!g' \
+            -e 's!^(\s*User)\s+daemon\s*$!\1 www-data!g' \
+            -e 's!^(\s*Group)\s+daemon\s*$!\1 www-data!g' \
+            "${{targets.destdir}}"/etc/httpd.conf \
+            "${{targets.destdir}}"/etc/extra/httpd-ssl.conf;
```

Comment:
"sedit" changes are hard to read. and can fail silently.
it would be nice if we had some sane way to validate that the changes you intend to make actually were made.

```
update() {
  local cfg="$1" var="$2" val="$3" found=""
  sed -i "s!^(\s*$var)(\s+)\S+!\1\2$val!g' "$cfg" || { echo "failed update $cfg"; return 1; }
  found=$(sed -n "s!^\s*$var\s+(\S+).*!\1!p")
  [ "$val" = "$found" || { echo "tried to update $var to $val, but it didnt work. found $found"; return 1; }
}

for f in /etc/httpd.conf /etc/extra/httpd-ssl.conf; do
  cp "$f" /tmp/dist 
  update CustomLog /proc/self/fd/1
  update ErrorLog /proc/self/fd/2
  ...
  # just to show the  diff output
  diff -u /tmp/dist "$f"
done
```

its not great, but then you don't silently fail to make the edit you want.
alternatively, you could just use patch. I don't know if that would be brittle acrros updates or not.

  
}